backend/tracim_backend/lib/cleanup/cleanup.py

```diff
@@ -0,0 +1,339 @@
+            ContentShare.content_id == content.content_id
+        )
+        for share in shares:
+            logger.info(
+                self, "delete share {} from content {}".format(share.share_id, share.content_id)
+            )
+            self.safe_delete(share)
+
+        # INFO - G.M - 2019-12-11 - delete children of content
+        if recursively:
+            for children in content.get_children(recursively=recursively):
+                self.delete_content(children)
+
+        for revision in content.revisions:
+            deleted_contents.extend(self.delete_revision(revision))
+
+        logger.info(self, "delete content {}".format(content.content_id))
+        deleted_contents.append(content.content_id)
+        self.safe_delete(content)
+        return deleted_contents
+
+    def delete_workspace(self, workspace: Workspace) -> int:
+        """
+        Delete workspace and associated stuff
+        - all content in the workspace
+        - all roles
+        - all upload permission
+        - workspace agenda
+
+        :param workspace: workspace to delete
+        :return: workspace_id of deleted workspace
+        """
+
+        # INFO - G.M - 2019-12-11 - delete role on workspace
+        roles = self.session.query(UserRoleInWorkspace).filter(
+            UserRoleInWorkspace.workspace_id == workspace.workspace_id
+        )
+        for role in roles:
+            logger.info(
+                self,
+                "delete role for user {} of workspace {}".format(
+                    role.user_id, workspace.workspace_id
+                ),
+            )
+            self.safe_delete(role)
+
+        # INFO - G.M - 2019-12-11 - delete permissions on workspace
+        upload_permissions = self.session.query(UploadPermission).filter(
+            UploadPermission.workspace_id == workspace.workspace_id
+        )
+        for upload_permission in upload_permissions:
+            logger.info(
+                self,
+                "delete upload_permission {} of workspace".format(
+                    upload_permission.upload_permission_id, workspace.workspace_id
+                ),
+            )
+            self.safe_delete(upload_permission)
+
+            logger.info(self, "delete workspace {}".format(workspace.workspace_id))
+            self.safe_delete(workspace)
+            for content in workspace.contents:
+                self.delete_content(content)
+        workspace_id = workspace.workspace_id
+        self.safe_delete(workspace)
+        return workspace_id
+
+    def delete_workspace_agenda(self, workspace_id: int) -> typing.Optional[str]:
+        # INFO - G.M - 2019-12-11 - delete workspace agenda
+        if not self.app_config.CALDAV__ENABLED:
+            logger.info(
+                self,
+                "will not delete workspace {} agenda, because caldav feature is disabled".format(
+                    workspace_id
+                ),
+            )
+            return
```

Comment:
I'm not sure about this. We should delete agenda user data even if radicale disabled.

Reply:
Radicale is a process, CALDAV_ENABLED is a config parameter to explicit caldav feature is enabled. We don't need radicale to be run to run this script, but if we do accept to try deleting in case caldav is disabled, we can have some trouble because we do not expect radicale config to be valid (path, etc...) in case radicale is disabled.

Reply:
done.